src/execution/__tests__/lists-test.ts

```diff
@@ -64,6 +71,146 @@ describe('Execute: Accepts any iterable as list value', () => {
   });
 });
 
+describe('Execute: Accepts async iterables as list value', () => {
+  function complete(rootValue: unknown) {
+    return execute({
+      schema: buildSchema('type Query { listField: [String] }'),
+      document: parse('{ listField }'),
+      rootValue,
+    });
+  }
+
+  function completeObjectList(
+    resolve: GraphQLFieldResolver<{ index: number }, unknown>,
+  ): PromiseOrValue<ExecutionResult> {
+    const schema = new GraphQLSchema({
+      query: new GraphQLObjectType({
+        name: 'Query',
+        fields: {
+          listField: {
+            resolve: async function* listField() {
+              yield await Promise.resolve({ index: 0 });
+              yield await Promise.resolve({ index: 1 });
+              yield await Promise.resolve({ index: 2 });
+            },
+            type: new GraphQLList(
+              new GraphQLObjectType({
+                name: 'ObjectWrapper',
+                fields: {
+                  index: {
+                    type: GraphQLString,
+                    resolve,
+                  },
+                },
+              }),
+            ),
+          },
+        },
+      }),
+    });
+    return execute({
+      schema,
+      document: parse('{ listField { index } }'),
+    });
+  }
+
+  it('Accepts an AsyncGenerator function as a List value', async () => {
+    async function* listField() {
+      yield await Promise.resolve('two');
```

Comment:
just a nit, could we remove `await Promise.resolve()` as an `async [generator] function` always returns/yields a Promise. it also does not schedule additional microtasks.

```js
// yields the same 
async function* foo() {
	yield await Promise.resolve(1)
	yield Promise.resolve(1)
	yield 1
}
```

Reply:
I added them because of the lint rule.

```
Async generator function 'listField' has no 'await' expression.eslint@typescript-eslint/require-await
```

I don't mind changing it to ignore the lint rule if that's preferred.